packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -105,6 +77,28 @@ const makeOuterKit = inner => {
  * @property {(oldDebt: Amount, oldCollateral: Amount, vaultId: VaultId) => void} updateVaultPriority
  */
 
+/**
+ * @typedef {Readonly<{
+ * assetNotifier: Notifier<import('./vaultManager').AssetState>,
+ * idInManager: VaultId,
+ * priceAuthority: ERef<PriceAuthority>,
+ * runMint: ZCFMint,
+ * vaultSeat: ZCFSeat,
+ * zcf: ContractFacet,
+ * }>} ImmutableState
```

Comment:
I wonder if we should have `props` a la React for immutable and have separate storage for those!

Reply:
Neat idea. We'd need the virtual objects API to send them in. Could be useful for distinguishing between things the contract gets to change vs things it just needs to be provided consistently.